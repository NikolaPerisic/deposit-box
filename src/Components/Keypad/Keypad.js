import React from "react";
import { keypads } from "../../bindings/keypads";
import "./Keypad.scss";

/*
Keypad component, gets all props from parent, gets active item
from props for animation of buttons
*/
const keypad = props => {
  const items = keypads.map((el, ind) => {
    return (
      <div
        className={
          props.activeItem === el
            ? "panel__keypads__item panel__keypads__item_active"
            : "panel__keypads__item"
        }
        key={`${el}-${ind}`}
        onClick={() => props.clickInput(el)}
      >
        {el}
      </div>
    );
  });
  return <div className="panel__keypads">{items}</div>;
};

export default keypad;
