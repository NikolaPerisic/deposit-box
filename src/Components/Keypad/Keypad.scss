$content1: "\25b2";
$content2: "\25c0";
$content3: "\25b6";
$content4: "\25bc";
$content5: "A";
$content6: "B";

.panel__keypads {
  grid-area: b;
  display: grid;
  grid-gap: 1rem;
  margin: 0 2rem;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(4, 1fr);
  text-align: center;

  .keypads--item {
    background-color: #63636e;
    color: #f3f3f3;
    touch-action: manipulation;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    font-size: 3em;
    font-weight: 500;
    border-radius: 3px;
    box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.4);
  }
  :nth-child(2n)::after {
    content: "";
    font-size: 0.7rem;
    font-weight: 600;
    color: #434343;
    position: absolute;
    bottom: 2px;
    right: 3px;
  }
  :nth-child(2):after {
    content: $content1;
  }
  :nth-child(4):after {
    content: $content2;
  }
  :nth-child(6):after {
    content: $content3;
  }
  :nth-child(8):after {
    content: $content4;
  }
  :nth-child(10):after {
    content: $content5;
  }
  :nth-child(12):after {
    content: $content6;
  }
  .keypads--item__active {
    box-shadow: none;
    color: #f1c40f;
  }
}
