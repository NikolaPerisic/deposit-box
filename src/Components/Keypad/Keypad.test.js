import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Keypad from "./Keypad";

describe("Keypad Component", () => {
  const props = {
    activeItem: null
  };
  //
  it("renders withour errors", () => {
    const KeypadComponent = shallow(<Keypad {...props} />);
    expect(toJson(KeypadComponent)).toMatchSnapshot();
  });
});
