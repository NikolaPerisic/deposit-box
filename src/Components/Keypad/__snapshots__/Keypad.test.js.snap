// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad Component renders withour errors 1`] = `
<div
  className="panel__keypads"
>
  <div
    className="panel__keypads__item"
    key="7-0"
    onClick={[Function]}
  >
    7
  </div>
  <div
    className="panel__keypads__item"
    key="8-1"
    onClick={[Function]}
  >
    8
  </div>
  <div
    className="panel__keypads__item"
    key="9-2"
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="panel__keypads__item"
    key="4-3"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="panel__keypads__item"
    key="5-4"
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="panel__keypads__item"
    key="6-5"
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="panel__keypads__item"
    key="1-6"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="panel__keypads__item"
    key="2-7"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="panel__keypads__item"
    key="3-8"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="panel__keypads__item"
    key="*-9"
    onClick={[Function]}
  >
    *
  </div>
  <div
    className="panel__keypads__item"
    key="0-10"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="panel__keypads__item"
    key="L-11"
    onClick={[Function]}
  >
    L
  </div>
</div>
`;
